File: frontend/src/types/reservation.ts
export interface Reservation {
    id: number;
    userId: number;
    roomId: number;
    date: string; // YYYY-MM-DD
    startSlot: number;
    endSlot: number;
    status: string;
    checkinTime: string | null; // or Date?
    createdAt: string; // or Date?
}

File: backend/src/main/java/kr/ac/hanyang/backend/dto/Reservation.java
package kr.ac.hanyang.backend.dto;

import lombok.Getter;
import lombok.Setter;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
public class Reservation {
    private int id;
    private int userId;
    private int roomId;
    private LocalDate date;
    private int startSlot;
    private int endSlot;
    private String status;
    private LocalDateTime checkinTime;
    private Boolean checkinRequired;  // 체크인 필요 여부
    private LocalDateTime createdAt;
}
